src/fsharp/FSharp.Core/array.fs

```diff
@@ -1270,6 +1270,96 @@ namespace Microsoft.FSharp.Collections
                 let count' = Operators.min count len
                 Microsoft.FSharp.Primitives.Basics.Array.subUnchecked 0 count' array
 
+        [<CompiledName("RemoveAt")>]
+        let removeAt (index: int) (source: 'T[]) : 'T[] =
+            checkNonNull "source" source
+            if index < 0 || index >= source.Length then invalidArg "index" "index must be within bounds of the array"
+            
+            let length = source.Length - 1
+            let result = Microsoft.FSharp.Primitives.Basics.Array.zeroCreateUnchecked length
+
+            for i in 0..length do
```

Comment:
I don't see why enumeration is needed, especially with conditional inside. Arrays have index based access. You just check the index, create new array and then do 2 Span/Buffer/Array (depends on what's available now in FSharl.Core) Copy calls and that's it. I would like you to change array implementations using this approach as the one you did simply does unnecessary things and thus not optimal.

Reply:
I updated the implementation to use array.copy